### PR TITLE
Replace old version tests

### PR DESCRIPTION
**Pull request by dnephin**
Remove tests of `/version` from the old integration suite and replace them with a test in the new suite.

### CODE REVIEW

File: integration/system/version_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+func TestVersion(t *testing.T) {
+	client, err := request.NewClient()
+	require.NoError(t, err)
+
+	version, err := client.ServerVersion(context.Background())
+	require.NoError(t, err)
+
+	assert.Equal(t, api.DefaultVersion, version.APIVersion)
```
Comment (vdemeester) [inline integration/system/version_test.go]: What if the daemon is being compiled at a different time (or with different version) than the source version being compiled ? Should we check only a `200` response code with valid version format instead ?

**Review by vdemeester — COMMENTED** [review]
SGTM but one question, mainly related to e2e

File: integration/system/version_test.go (hunk 1 above)
Comment (dnephin) [reply integration/system/version_test.go]: I don't think we should really expect the test suite to work with a different version of the daemon. 
This is what the old test did so at least it's consistent with what we were testing before.

Comment (dnephin) [reply integration/system/version_test.go]: I guess it is a problem with #34862

I can adjust this version check

**Comment by dnephin** [comment]
Ok, updated to only check `NotNil` for versions

**Review by vdemeester — APPROVED** [review]
LGTM ✌️

**Review by AkihiroSuda — APPROVED** [review]
LGTM if green

File: integration/system/main_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	testEnv.Print()
+	os.Exit(m.Run())
+}
+
+func setupTest(t *testing.T) func() {
```
Comment (AkihiroSuda) [inline integration/system/main_test.go]: where is this called?

**Review by boaz0 — APPROVED** [review]
💯

File: integration/system/main_test.go (hunk 2 above)
Comment (dnephin) [reply integration/system/main_test.go]: it's not called yet

**Comment by dnephin** [comment]
test failure is unrelated. it's caused by a config change to jenkins. good to merge?

**Pull request merged**